Update tracking.service.ts

trollkotze opened the pull request:
Bei der Übermittlung der Shopify-Orders an ParcelLab fehlt die Adresszeile 2.
Vielleicht ist nur diese Änderung nötig. Bin mir nicht sicher.

trollkotze committed:
Update tracking.service.ts

JumpLink commented:
@trollkotze can you check if `shopifyOrder.shipping_address.address2` is set and not falsy / empty?

Otherwise your changes will always add a new line, also if there is no  address2.

JumpLink commented on backend/src/tracking/tracking/tracking.service.ts:
```diff
@@ -667,7 +667,7 @@ export class ParcelLabTrackingService {
       recipient_notification: this.getName(shopifyOrder),
       statuslink: shopifyOrder?.order_status_url,
-      street: shopifyOrder?.shipping_address?.address1,
+      street: `${shopifyOrder?.shipping_address?.address1}\n${shopifyOrder?.shipping_address?.address2}`,
```
@trollkotze can you check if shopifyOrder.shipping_address.address2 is set and not falsy / empty?

Otherwise your changes will always add a new line, also if there is no address2.

trollkotze replied:
Okay.

JumpLink replied:
Same on line 585

trollkotze committed:
Update tracking.service.ts

Don't add second line to address if address2 is empty.

trollkotze committed:
Update tracking.service.ts

code formatting

JumpLink commented on backend/src/tracking/tracking/tracking.service.ts:
```diff
@@ -667,7 +667,12 @@ export class ParcelLabTrackingService {
       recipient_notification: this.getName(shopifyOrder),
       statuslink: shopifyOrder?.order_status_url,
-      street: shopifyOrder?.shipping_address?.address1 + (shopifyOrder?.shipping_address?.address2?.trim() ? `\n${shopifyOrder?.shipping_address?.address2}` : ''),
+      street: shopifyOrder?.shipping_address?.address1
```
Can you create a method `getStreet` for this? Because you do the same thing on two different parts. See `this.getName` above.

trollkotze committed:
Update tracking.service.ts

Add method to combine address into one field and use it for fulfillment and order.

trollkotze committed:
Update tracking.service.ts

fix typo
